src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -579,6 +583,7 @@ mongoc_cluster_run_command_monitored (mongoc_cluster_t *cluster,
                                          cmd->operation_id,
                                          &server_stream->sd->host,
                                          server_id,
+                                         is_redacted,
```

Comment:
Minor: I think there is one case where we over-redact. If the `hello` / `legacy hello` command includes `speculativeAuthenticate` but the server reply does not contain `speculativeAuthenticate`. We'll redact the reply regardless.

IMO it is very low impact. It would only affect users running `hello` with `speculativeAuthenticate` through the generic command helpers. Since it is low impact, I am OK with this behavior. I think it is worth noting that behavior in a comment here.

Reply:
Edit: Nevermind... the spec seems clear that the reply should always be redacted if the command was redacted

> Some commands and replies will contain sensitive data and in order to not risk the leaking of this data to external sources or logs their commands AND replies MUST be redacted from the events.

Reply:
Yeah, I had hoped to leave replies to `hello` unreacted if they don't contain any sensitive infos, but doing so would've required either changes in other driver, or having no test for it. I decided to always redact replies to sensitive hello commands here instead.